src/aotcompile.cpp

```diff
@@ -1655,6 +1657,8 @@ void jl_dump_native_impl(void *native_code,
                                      GlobalVariable::ExternalLinkage,
                                      data, "jl_system_image_data");
         sysdata->setAlignment(Align(64));
+        if (TheTriple.isX86() && TheTriple.isArch64Bit() && TheTriple.isOSLinux())
+            sysdata->setSection(".ldata");
```

Comment:
IIUC, this is already automatic, once you select Medium code model globally, although can also be opted for explicitly for each variable:
https://llvm.org/doxygen/TargetMachine_8cpp_source.html
```suggestion
        sysdata->setCodeModel(CodeModel::Large);
```

Reply:
It wasn't doing it automatically for me :(. Not sure why

Reply:
https://github.com/llvm/llvm-project/pull/72078 also the per global option was just added.

Reply:
I guess this should have an LLVM_VERSION ifdef here then?